File: prak python/Motorcycle.py
from Vehicle import Vehicle  # Jika Vehicle.py ada di dalam direktori yang sama dengan file ini

class Motorcycle(Vehicle):
    def __init__(
        self, license_plate, brand, production_year, color, motor_type, tank_capacity
    ):
        super().__init__(license_plate, brand, production_year, color)
        self._motor_type = motor_type
        self._tank_capacity = tank_capacity

    @property
    def motor_type(self):
        return self._motor_type

    @property
    def tank_capacity(self):
        return self._tank_capacity

    @motor_type.setter
    def motor_type(self, motor_type):
        self._motor_type = motor_type

    @tank_capacity.setter
    def tank_capacity(self, tank_capacity):
        self._tank_capacity = tank_capacity

File: prak cpp/Motorcycle.cpp
#pragma once

#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle {
private:
    string _motor_type;
    double _tank_capacity;

public:
    Motorcycle(string license_plate, string brand, int production_year, string color, string motor_type, double tank_capacity)
        : Vehicle(license_plate, brand, production_year, color), _motor_type(motor_type), _tank_capacity(tank_capacity) {}

    string get_motor_type() const {
        return _motor_type;
    }

    double get_tank_capacity() const {
        return _tank_capacity;
    }

    void set_motor_type(string motor_type) {
        _motor_type = motor_type;
    }

    void set_tank_capacity(double tank_capacity) {
        _tank_capacity = tank_capacity;
    }
};